apollo-router/src/plugins/connectors/http_json_transport.rs

```diff
@@ -77,43 +78,74 @@ pub(crate) fn make_request(
         &flat_inputs,
     )?;
 
-    let (json_body, body, apply_to_errors) = if let Some(ref selection) = transport.body {
+    let request = http::Request::builder()
+        .method(transport.method.as_str())
+        .uri(uri.as_str());
+
+    // add the headers and if content-type is specified, we'll check that when constructing the body
+    let (mut request, content_type) = add_headers(
+        request,
+        original_request.supergraph_request.headers(),
+        &transport.headers,
+        &flat_inputs,
+    );
+
+    let is_form_urlencoded = content_type.as_ref() == Some(&mime::APPLICATION_WWW_FORM_URLENCODED);
+
+    let (json_body, form_body, body, apply_to_errors) = if let Some(ref selection) = transport.body
+    {
         let (json_body, apply_to_errors) = selection.apply_with_vars(&json!({}), &inputs);
+        let mut form_body = None;
         let body = if let Some(json_body) = json_body.as_ref() {
-            hyper::Body::from(serde_json::to_vec(json_body)?)
+            if is_form_urlencoded {
+                let encoded = encode_json_as_form(json_body)
+                    .map_err(HttpJsonTransportError::FormBodySerialization)?;
+                form_body = Some(encoded.clone());
+                hyper::Body::from(encoded)
+            } else {
+                request = request.header(CONTENT_TYPE, mime::APPLICATION_JSON.essence_str());
```

Comment:
So, if they provided an explicit header which resolves to urlencoded form, we use that exact header. Otherwise, we replace whatever they gave us with `application/json`, right?

Definitely need a validation to make it clear that this will happen.

Reply:
yep that's exactly right. validating values is a good idea, though it'll be possible to bypass static validation with `value: "{$config.some_content_type_we_cant_see}"`